src/pkgs.rs

```diff
@@ -0,0 +1,116 @@
+#[allow(dead_code)]
+
+use crate::*;
+use std::vec::Vec;
+use std::process::Command;
+
+#[derive(Clone, Debug)]
+pub enum PkgManager {
+    Arch,
+    Debian,
+    Void,
+    Fedora,
+    BSD,
+    Solus,
+    Suse,
+    Alpine,
+    Gentoo,
+    Pip,
+    Cargo,
+    Unknown,
+
+    // TODO
+    //CRUX,
+    //KISS,
+    //Slackware,
+    //NixOS,
+    //Bedrock,
+}
+
+pub struct PkgInfo {
+    manager: Vec<PkgManager>,
+    count:   usize,
+}
+
+impl PkgInfo {
+    pub fn new() -> PkgInfo {
+        PkgInfo {
+            manager: Vec::new(),
+            count:   0,
+        }
+    }
+
+    pub fn get(&mut self) -> Result<()> {
+        for manager in self.manager.clone() {
+            let output = match manager {
+                PkgManager::Arch    => Command::new("pacman").arg("-Qq")
+                    .output().context(Pkgcount)?,
+                PkgManager::Debian  => Command::new("apt").arg("list")
+                    .output().context(Pkgcount)?,
+                PkgManager::Void    => Command::new("xbps-query").arg("-l")
+                    .output().context(Pkgcount)?,
+                PkgManager::Fedora  => Command::new("dnf").arg("list --installed")
+                    .output().context(Pkgcount)?,
+                PkgManager::BSD     => Command::new("pkg").arg("info")
+                    .output().context(Pkgcount)?,
+                PkgManager::Suse    => Command::new("rpm").arg("-qa")
+                    .output().context(Pkgcount)?,
+                PkgManager::Solus   => Command::new("eopkg").arg("list-installed")
+                    .output().context(Pkgcount)?,
+                PkgManager::Alpine  => Command::new("apk").arg("info")
+                    .output().context(Pkgcount)?,
+                PkgManager::Gentoo  => Command::new("qlist").arg("-I")
```

Comment:
This can be found better as I stated in https://github.com/rsfetch/rsfetch/issues/51#issuecomment-565910597. On top of that, it'll be one less extra executable to be required or called. You don't need to do this right now or anything, just thought I'd bring it to your attention. I can always do it later as well, if you don't want to. I've just been hesitant on touching any of the files while you were still going through and changing/refactoring things.

Reply:
This can be done in another release, I think. Anyway, I'd prefer to complete this in another PR, since this one was supposed to be solely a refactor, not changing anything in the code that wasn't really necessary.

Reply:
Yep, got it. Resolving.